src/Makefile.am

```diff
@@ -136,9 +136,18 @@ BITCOIN_CORE_H = \
   clientversion.h \
   coins.h \
   common/args.h \
+  common/asmap.h \
+  common/bip32.h \
   common/bloom.h \
+  common/bytevectorhash.h \
```

Comment:
I think for simple utilities like the `bytevectorhash` module, I kind of agree, since it does not pull in further unwanted deps, nor is used for anything opinionated (and similarly `readwritefile.h`). `sock.h` however includes `compat.h`, which we've previously worked to remove from the kernel header tree, so I think I'll leave that one as is.